app/assets/javascripts/stimulus/loaders/autoloader.js

```diff
@@ -3,19 +3,32 @@ import { Application } from "stimulus"
 const application = Application.start()
 
 function autoload() {
-  Array.from(document.querySelectorAll('[data-controller]')).forEach((element) => {
-    const controllerNames = element.attributes["data-controller"].value.split(" ")
+  for (const element of document.querySelectorAll("[data-controller]")) {
+    autoloadControllers(element)
+  }
+}
 
-    controllerNames.forEach((controllerName) => {
-      let controllerFilename = `${controllerName}_controller`
+function autoloadControllers(element) {
+  const controllerTokens = element.getAttribute("data-controller") || ""
```

Comment:
You can remove `|| ""` because getAttribute will return empty string if we have HTML `<div data-controller>`.
Also everywhere we already select elements with data-controller attribute